test/e2e/aadpodidentity_test.go

```diff
@@ -194,11 +194,32 @@ var _ = Describe("Kubernetes cluster using aad-pod-identity", func() {
 	It("should delete the AzureAssignedIdentity if the deployment is deleted", func() {
 		setUpIdentityAndDeployment(keyvaultIdentity, "", "1")
 
+		podName, err := pod.GetNameByPrefix(identityValidator)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(podName).NotTo(Equal(""))
+
+		// Get the name of the node to assign the identity to
+		nodeName, err := pod.GetNodeName(podName)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(nodeName).NotTo(Equal(""))
+
+		n, err := node.Get(nodeName)
+		Expect(err).NotTo(HaveOccurred())
+
 		waitForDeployDeletion(identityValidator)
 
+		// Check for AzureAssignedIdentities.
 		ok, err := azureassignedidentity.WaitOnLengthMatched(0)
 		Expect(err).NotTo(HaveOccurred())
 		Expect(ok).To(Equal(true))
+
+		resourceManager, err := getResourceManager(n)
+		Expect(err).NotTo(HaveOccurred())
+
+		userAssignedIdentities, err := resourceManager.GetUserAssignedIdentities()
+		Expect(err).NotTo(HaveOccurred())
+		// Check if the user assigned Identity has been removed from the underlying VM/VMSS.
+		Expect(len(userAssignedIdentities)).To(Equal(0))
```

Comment:
Instead of checking the length, can we just check the `keyvaultIdentity` is not in the map?

Reply:
Adding a note here based on our discussion offline - get the list of identities on the node before the test is run. After the test is run, get the list of identities again from the node and check the list is same as what it was at the start of the test.

- Helps validate the identity that was assigned as part of the test has been successfully removed.
- Ensures no new identities were assigned by MIC